lib/pos_consensus.go

```diff
@@ -302,13 +302,6 @@ func (fc *FastHotStuffConsensus) handleBlockProposalEvent(
 		sendMessageToRemoteNodeAsync(validator, blockProposal)
 	}
 
-	// Broadcast the block to all inbound non-validator peers. This allows them to sync
```

Comment:
The block propoer's steady state FastHotStuffConsensus shouldn't broadcast blocks to non-validators. Instead, all non-validator nodes should use the existing Inv message to gossip known block hashes to each other to propagate blocks across the network